src/main/solution_space/solution_graph.py

```diff
@@ -228,3 +231,32 @@ def add_code_info_chain(self, code_info_chain: Optional[List[Tuple[Code, CodeInf
                 prev_vertex.add_child(next_vertex)
                 prev_vertex = next_vertex
         log.info(f'Finish adding code-user chain')
+
+    @staticmethod
+    def update_ages_and_experiences(code_info_list: List[CodeInfo],
```

Comment:
I don't fully understand the sense of this method. All two times that it's called, you pass en empty lists for ages and experiences, so why don't you want to get ages and experiences directly, without passing any lists? Also, if these lists were not empty, they would contain some incorrect (or doubled) elements, because you don't check what they contain